### PR TITLE
Email validation

### PR DESCRIPTION
**Pull request by skarbat**
Adds email validation during the registration process

 * Added tests for malformed and correct email addresses
 * Implemented function to validate email
 * Implemented validation control in the Gtk GUI side
 * Added spanish translations

Tested on the PI and seems to do the job @tombettany @Ealdwulf

### CODE REVIEW

File: po/es_AR.po [hunk 1]
```diff
@@ -4,18 +4,16 @@
...
+"Report-Msgid-Bugs-To: Kano <dev@kano.me>\n"
 "POT-Creation-Date: 2016-09-23 16:28+0100\n"
-"PO-Revision-Date: YEAR-MO-DA HO:MI+ZONE\n"
-"Last-Translator: FULL NAME <EMAIL@ADDRESS>\n"
-"Language-Team: LANGUAGE <LL@li.org>\n"
-"Language: \n"
+"PO-Revision-Date: 2016-09-29 16:28+0100\n"
+"Last-Translator: Kano <dev@kano.me>\n"
+"Language-Team: Spanish Team <dev@kano.me>\n"
+"Language: Argentinian\n"
```
Comment (tombettany) [inline po/es_AR.po]: Should this be `es_AR`?

File: po/es_AR.po [hunk 2]
```diff
@@ -4,18 +4,16 @@
 # FIRST AUTHOR <EMAIL@ADDRESS>, YEAR.
 #
 
-#, fuzzy
-
```
Comment (tombettany) [inline po/es_AR.po]: Are we sure that this bit isn't needed?

Comment (skarbat) [inline po/es_AR.po]: not sure Tom, `make` will complain otherwise, why are we carrying this I don't know, I'll look at it

File: po/es_AR.po (hunk 1 above)
Comment (skarbat) [inline po/es_AR.po]: Yep - fixed.

File: po/es_AR.po (hunk 2 above)
Comment (skarbat) [inline po/es_AR.po]: Apparently we should be regnerating the messages file in the first place, but I think it's harmless:

 * https://www.gnu.org/software/gettext/manual/html_node/Fuzzy-Entries.html#Fuzzy-Entries

**Comment by skarbat** [comment]
[[World Registration] Restrict special characters in login](https://trello.com/c/gxUNBbb1/394-world-registration-restrict-special-characters-in-login)

**Comment by tombettany** [comment]
Due to concern for this late change and the fact that it isn't a huge bug, let's hold off on this one until after release.

**Comment by Ealdwulf** [comment]
I guess this will eventually need to support punycode. But looks good for the time being 👍

**Comment by skarbat** [comment]
Good point @Ealdwulf , also we should put it in common with the Greeter as well.
Merging now for testing tomorrow.

**Pull request merged**